File: src/main/scala/com/thoughtworks/pli/remotepair/idea/dialogs/SyncProgressDialog.scala
package com.thoughtworks.pli.remotepair.idea.dialogs

import com.thoughtworks.pli.remotepair.core.models.{MyProject, MyIde}
import com.thoughtworks.pli.remotepair.core.ui.VirtualComponents._
import com.thoughtworks.pli.remotepair.core.ui.dialogs.VirtualSyncProgressDialog
import com.thoughtworks.pli.remotepair.idea.listeners.PairEventListeners

import SwingVirtualImplicits._

import scala.language.reflectiveCalls

class SyncProgressDialog(val currentProject: MyProject, val myIde: MyIde, val pairEventListeners: PairEventListeners)
  extends _SyncProgressDialog with JDialogSupport with VirtualSyncProgressDialog {

  override val dialog: VirtualDialog = this
  override val closeButton: VirtualButton = _closeButton
  override val progressBar: VirtualProgressBar = _progressBar
  override val messageLabel: VirtualLabel = _messageLabel

  init()

}

File: src/main/scala/com/thoughtworks/pli/remotepair/idea/dialogs/_SyncProgressDialog.java
package com.thoughtworks.pli.remotepair.idea.dialogs;

import javax.swing.*;

public class _SyncProgressDialog extends JDialog {
    private JPanel contentPanel;
    protected JProgressBar progressBar;
    protected JLabel messageLabel;
    protected JButton closeButton;

    public _SyncProgressDialog() {
        setContentPane(contentPanel);
    }

}
